MonoGame.Framework/GameWindow.cs

```diff
@@ -144,9 +144,10 @@ protected void OnScreenDeviceNameChanged ()
 		}
 
 #if WINDOWS || WINDOWS_UAP || DESKTOPGL || ANGLE
-		protected void OnTextInput(object sender, TextInputEventArgs e)
-		{
-            EventHelpers.Raise(this, TextInput, e);
+        internal void OnTextInput(object sender, TextInputEventArgs e)
+		{
+            if (TextInput != null)

```

Comment:
Now that TextInputEventArgs is a struct, I can see not wanting to pass it to another method to raise it because that would be another unnecessary copy.  If you're going to raise events manually, the best practice is to store event in a local variable before checking for null and invoking it.  The reason is because it's possible (while highly unlikely) that another thread could be activated between the check for null and the actual invocation that will remove the last subscriber to the event.  This would lead to a hard to duplicate NullReferenceException.

If you're allowed to use C#6 language features, then the simple way around this is to do the following.
`TextInput?.Invoke(sender, e)`.

If C#6 is not an option, then using a local variable is the way to go.

Side note:  Nice catch here.  Looks like there was a hidden bug where the sender should have been passed to the EventHelpers.Raise method.